lib/common.js

```diff
@@ -593,6 +593,22 @@ function deepEqual(expected, actual) {
   return expected === actual
 }
 
+const timeouts = []
+
+function addTimeout(id) {
```

Comment:
I think it would clean up the calling code if, instead of taking a timeout ID, we had a `setTimeout` function that wrapped the native call and saved the id. eg:
```js
function setTimeout(callback, delay, ...args) {
  const id = timers.setTimeout(callback, delay, ...args)
  timeouts.push(id)
  return id
}
```

Reply:
if that is the approach you would like to take, i would also like to make the suggestion of adding our own timers module so we are able to follow the same pattern for all timers and not just setTimeout, but also setInterval(maybe used in the future) and setImmediate. As This will provide the opportunity to cover similar uses cases moving forward with less code changes. Dont think doing the same for nextTick is needed as there is no clear method.